### PR TITLE
Update bdk version to 0.23.0

### PR DESCRIPTION
**Pull request by rajarshimaitra**
### Description

Candidate PR for bdk version update.. 

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk-cli/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -81,4 +83,4 @@ regtest-bitcoin = ["regtest-node" , "rpc"]
 regtest-electrum = ["regtest-node", "electrum", "electrsd/electrs_0_8_10"]
 #TODO: Check why esplora in electrsd isn't working.
 #regtest-esplora-ureq = ["regtest-node", "esplora-ureq", "electrsd/esplora_a33e97e1"]
-#regtest-esplora-reqwest = ["regtest-node", "esplora-reqwest", "electrsd/esplora_a33e97e1"]
+#regtest-esplora-reqwest = ["regtest-node", "esplora-reqwest", "electrsd/esplora_a33e97e1"]
```
Comment (danielabrozzoni) [inline Cargo.toml:86]: I wonder why your editor removes newlines at the end of the files, while mine adds them automatically 😂

Comment (rajarshimaitra) [reply Cargo.toml]: Ah thanks.. I don't think its my editor.. I probably did it myself by mistake.. Reverted..

**Comment by rajarshimaitra** [comment]
@notmandatory will update this once https://github.com/bitcoindevkit/bdk-reserves/pull/12 goes in..

**Comment by notmandatory** [comment]
Renamed this to "Update bdk version to 0.23.0"

**Review by notmandatory — APPROVED** [review]
utACK 536f65fd8efca21886bf817e845f322896a5cc6c

**Comment by rajarshimaitra** [comment]
All green.. Merged..

**Pull request merged**